test/es6/toPrimitive.js

```diff
@@ -159,6 +215,283 @@ var tests = [
+                assert.throws(()=>(2+o), TypeError, "2+o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o+'abc'), TypeError, "o+'abc'", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>('abc'+o), TypeError, "'abc'+o", "[Symbol.toPrimitive]: invalid argument");
+
+                //
+                // abstract relational comparison
+                //
+
+                assert.throws(()=>(o<1), TypeError, "o<1", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(1<o), TypeError, "1<o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o<=25), TypeError, "o<=25", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(-9<=o), TypeError, "-9<=o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o>1), TypeError, "o>1", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o>=25), TypeError, "o>=25", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(-9>=o), TypeError, "-9>=o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o<o), TypeError, "o<o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o<=o), TypeError, "o<=o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o>o), TypeError, "o>o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o>=o), TypeError, "o>=o", "[Symbol.toPrimitive]: invalid argument");
+
+                //
+                // abstract equality comparison
+                //
+
+                assert.isTrue(o==o, "o==o should not call ToPrimitive");
+                assert.throws(()=>('abc'==o), TypeError, "'abc'==o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o=='abc'), TypeError, "o=='abc'", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(100==o), TypeError, "100==o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o==100), TypeError, "o==100", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(Symbol()==o), TypeError, "Symbol()==o", "[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(o==Symbol()), TypeError, "o==Symbol()", "[Symbol.toPrimitive]: invalid argument");
+
+                //
+                // Date.prototype[@@toPrimitive]
+                //
+
+                assert.throws(()=>(Date.prototype[Symbol.toPrimitive].call(o)), TypeError, "Date.prototype[@@toPrimitive] hint==undefined", "Date[Symbol.toPrimitive]: invalid argument");
+                assert.throws(()=>(Date.prototype[Symbol.toPrimitive].call(o, {})), TypeError, "Date.prototype[@@toPrimitive] hint=={}", "Date[Symbol.toPrimitive]: invalid argument");
+            });
+        }
+    },

```

Comment:
Will make a separate PR for SIMD  type if necessary.

---
In reply to: [71951385](https://github.com/Microsoft/ChakraCore/pull/1319#discussion_r71951385) [](ancestors = 71951385)